### ISSUE TITLE
Refactor Tagging System

### ISSUE THREAD

**Issue by bbertucc** [bbertucc/equalify#262]
Tags are added to occurrences vs Rules. That incorrectly relates tags to occurrences. Example: If an occurrence has the `WCAG 2.1` tag from one rule violation and `Best Practice` from another, and I resolve the `Best Practice` violation, the occurrence keeps the tag.

**Comment by bbertucc** [comment]
We could just update the tagging system to resolve tags, but that would create a lot more complexity at this point than updating the data structure. Adding notes around that task here: https://github.com/EqualifyEverything/equalify/issues/257

**Comment by bbertucc** [comment]
closing this in favor of #257
